### PR TITLE
Added navigation for mobile devices

### LINKED ISSUE #2196: feat: add new design of nav menu in mobile devices
> You can view the figma file [here](https://www.figma.com/file/ntisIxc9QHwOZ828CH5G3a/LinksHub?type=design&node-id=17-231&mode=design&t=29U3Qw0LktnVRteR-0).

image

> The design should be responsive.

---

ℹ️ We recommend updating the styling of existing components, otherwise you can add extra components if there is a need.

- You can get icons from [here](https://www.figma.com/file/ntisIxc9QHwOZ828CH5G3a/LinksHub?node-id=312%3A5847&mode=dev)
- You need to implement the active state and navigating to correct route based on selected nav icon.

> Any changes to the above design are welcome. Please feel free to share your thoughts, and we will discuss them thoroughly.

---
### ⚠️ Make sure to push all the changes into `dev` branch

### PR DESCRIPTION
**Pull request by MAXASSASSINS**
## Fixes Issue 

closes #2196

## Changes proposed

- Created a new component handling navigation in mobile devices.
- Added some css in general layout.
- Fixed the width of the linkshub description on home page.

## Screenshots
Screenshot 2023-12-01 at 1 42 53 AM
  
Screenshot 2023-12-01 at 1 44 41 AM

### CODE REVIEW

**Review by CBID2 — APPROVED** [review]
Looks good to me! @Anmol-Baranwal, @rupali-codes, check this out! 😁

File: components/MobileBottomNav/MobileBottomNav.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            >
+              <span className="flex items-center justify-center">
+                {isActive ? activeIcon : inActiveIcon}
+              </span>
+              <span className="text-xs">{label}</span>
+            </Link>
+          ) : (
+            <div
+              className={`w-full flex flex-col px-4 p-3 gap-2 font-medium rounded-xl hover:bg-slate-100 hover:bg-opacity-50 dark:hover:bg-zinc-400 dark:hover:bg-opacity-10 ${
+                isActive ? 'text-primary dark:text-white' : 'text-gray-400'
```
Comment (aftabrehan) [inline components/MobileBottomNav/MobileBottomNav.tsx:73]: These classes are repeated, can we extract this code?

File: layouts/GeneralLayout.tsx [hunk 2]
```diff
@@ -20,7 +21,7 @@ const GeneralLayout = ({ children }: { children: ReactNode }) => {
       <Header />
       <SideNavbar />
       <div
-        className={`row-start-2 row-end-3 min-h-[100vh-72px] w-full bg-gray-100 dark:bg-[#101623] ${
+        className={`row-start-2 row-end-3 overflow-y-scroll max-h-[calc(100vh-156px)] sm:max-h-[calc(100vh-72px)] w-full bg-gray-100 dark:bg-[#101623] ${
```
Comment (aftabrehan) [inline layouts/GeneralLayout.tsx:24]: Could you please explain this change?

File: components/MobileBottomNav/MobileBottomNav.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    {
+      inActiveIcon: <SaveInActiveIcon className={inActiveIconCls} />,
+      activeIcon: <SaveActiveIcon className={activeIconCls} />,
+      label: 'Saved',
+      href: '/saved',
+    },
+    {
+      inActiveIcon: <SearchInAciveIcon className={inActiveIconCls} />,
+      activeIcon: <SearchAciveIcon className={activeIconCls} />,
+      label: 'Search',
```
Comment (aftabrehan) [inline components/MobileBottomNav/MobileBottomNav.tsx:38]: We need to pass `onClick` function from here. 

so we can do:
```js
onClick 
   ? <button onClick={onClick} className={commonCls}>{...}</button>
   : <Link href={href} className={commonCls}>{...}</Link>
```

This approach is much better and simpler. What do you think?

**Review by aftabrehan — CHANGES_REQUESTED** [review]
Excellent PR, @MAXASSASSINS 💪 🚀 💯 

It just needs minor adjustments and then we're good to go 🚀

**Review by Anmol-Baranwal — APPROVED** [review]
@MAXASSASSINS 

🚀 Looks good to me. 

> Thanks! It is deployed.

**Review by aftabrehan — APPROVED** [review]
Looks good to me! 👍

**Comment by Anmol-Baranwal** [comment]
@rupali-codes 
Review and merge this :)

**Review by rupali-codes — APPROVED** [review]
looks perfect

**Pull request merged**